src/sequence.lisp

```diff
@@ -32,6 +32,25 @@
     (t
      (not-seq-error seq))))
 
+(defun string-reverse (string)
+  (let* ((length (length string))
+         (new-string (make-string length)))
+    (dotimes (index length new-string)
+      (setf (aref new-string index) (aref string (- length (1+ index)))))))
+
+(defun vector-reverse (vector)
```

Comment:
It would be much better to to unify (a string is a vector after all) them, didn't think about `array-element-type`.